cmd/kubelet/app/server.go

```diff
@@ -367,6 +373,32 @@ func UnsecuredDependencies(s *options.KubeletServer, featureGate featuregate.Fea
 	if err != nil {
 		return nil, err
 	}
+	var tp oteltrace.TracerProvider
+	if utilfeature.DefaultFeatureGate.Enabled(features.KubeletTracing) {
+		hostname, err := nodeutil.GetHostname(s.HostnameOverride)
+		if err != nil {
+			return nil, err
+		}
+		opts := []otlpgrpc.Option{}
+		resourceOpts := []otelsdkresource.Option{
+			otelsdkresource.WithAttributes(
+				semconv.ServiceNameKey.String(hostname),
+				semconv.ServiceInstanceIDKey.String(componentKubelet),
+			),
+		}
+		if s.KubeletConfiguration.Tracing.Endpoint != nil {
+			opts = append(opts, otlpgrpc.WithEndpoint(*s.KubeletConfiguration.Tracing.Endpoint))
+		}
+		opts = append(opts, otlpgrpc.WithInsecure())
+		tc := &configv1alpha1.TracingConfiguration{}
+		if err := configv1alpha1.Convert_config_TracingConfiguration_To_v1alpha1_TracingConfiguration(&s.KubeletConfiguration.Tracing, tc, nil); err != nil {
```

Comment:
Is it necessary to do this conversion explicitly? I don't think I've seen anything like this in a config API...

/cc @liggitt

Reply:
updated to component-base/config/v1alpha1 TracingConfiguration everywhere

Reply:
i think i have it sorted out now, ptal

Reply:
ptal again, thank you